### PR TITLE
Add Intel GPU (XPU) support 

### PR DESCRIPTION
**Pull request by rajeshgangireddy**
Hi, Thanks for the great work with OpenFold3. 

**Summary**
This PR adds support for Intel GPUs for OpenFold3. Mostly it is based on two existing non-CUDA device patterns already in the codebase (ROCm and MPS).
Validated end-to-end  against an Intel Arc Pro B70 (XPU) and an NVIDIA RTX 3090 (CUDA) side by side, across 6 supported input modalities (protein, DNA, RNA, homomer, multimer, protein-ligand) for inference as well as training on a small subset of PDB. 

**Main Changes**

- Added XPUAccelerator for lightning (taken mostly from our other project [Anomalib project](https://github.com/open-edge-platform/anomalib/blob/5012f33619253d54c906d55550e8ce6d5598ddbe/src/anomalib/engine/accelerator/xpu.py#L12) 
- Added xpu support for helpers methods for emptying cache `empty_device_cache()`, and torch synchronisation 
- Device-agnostic core fixes: A few places still had hardcoded cuda for  `torch.amp.autocast("cuda", ...)`. Replaced it with   `autocast_device_type()` helper 
- Added `openfold3/entry_points/validate_xpu.py` based on `openfold3/entry_points/validate_rocm.py`
- pixi.toml: new openfold3-xpu pixi environment (similar to openfold3-rocm7): torch installed from download.pytorch.org/whl/xpu, no DeepSpeed/cuEquivariance. 
-  New XPU snapshots for the two Triton-triangle-kernel test modules above, regenerated on real Arc Pro B70 hardware.
- Docs: Added Intel XPU related installation and inference steps in `docs/source/Installation.md` and `docs/source/inference.md` respectively. Intel XPU sections mirroring the existing ROCm/MPS content (environment setup, Triton install, validate-openfold3-xpu)

**Related Issues**
None 

**Hardware Used** : 
-  Tested on  [Intel Arc B70](https://www.intel.com/content/www/us/en/products/sku/245797/intel-arc-pro-b70-graphics/specifications.html) as well as on [Intel Arc B60](https://www.intel.com/content/www/us/en/products/sku/243916/intel-arc-pro-b60-graphics/specifications.html) discrete GPUs. 
- PyTorch XPU also covers the integrated GPUs that Intel CPUs have.  So, I also tested inference on an iGPU on [Intel's Core Ultra 7 Processor 356H](https://www.intel.com/content/www/us/en/products/sku/245532/intel-core-ultra-7-processor-356h-18m-cache-up-to-4-70-ghz/specifications.html). 

**Testing** 
- `pixi run -e openfold3-xpu pytest openfold3/tests -m "not slow"`: 1032 passed, 28 skipped, 0 failed.
- `pixi run -e openfold3-cuda12 pytest openfold3/tests -m "not slow"`: 1055 passed, 5 skipped, 0 failed.
- Added TestAcceleratorWillUseXpu   and covers the auto-detection logic: XPU resolves correctly with/without CUDA present, and CUDA priority is preserved for "auto"/"gpu" when both accelerators exist on one box.
- Triton triangle-attention / triangle-multiplicative-update kernels produce numerically correct results on Intel XPU 
- All other linting / formatting checks pass 
- Accuracy validation on six modalities (protein, protein-ligand, DNA, homomer, multimer, RNA) similar to #336 
- Compared B70 inference vs RTX 3090 on the same machine shows near bit-identical between CUDA and XPU (≤0.04 Å backbone RMSD). Let me know if I need to post the details. 
- Training validation : Training (forward + backward + optimizer step + validation + checkpointing) was validated end-to-end on a single XPU device and cross-checked against single-device CUDA, using a small local PDB subset (8 train / 4 validation structures). After each epoch, train/loss, val/loss and val/rmsd on intel GPU are same as the ones on a CUDA (RTX 3090). 

**Other Notes**
- I see `torch.cuda.manual_seed_all` used in openfold3/core/utils/callbacks.py and so i had to use `torch.xpu.manual_seed_all(rank_specific_seed)`. But maybe it is better to just use `torch.manual_seed()`? 
- DeepSpeed EvoAttention and cuEquivariance triangle kernels remain NVIDIA-CUDA-only upstream (same situation as ROCm today); XPU training falls back to the stock/Triton attention path.
- No CI changes as there are no Intel GPU CI runners  

Let me know if I need to provide any further data or need to make some changes. 
Thanks

### CODE REVIEW

**Comment by jandom** [comment]
@rajeshgangireddy oh wow this is EXCITING – thank you so much! If it's not too much trouble to ask, do you know if this CPU or equivalent is available though some hyperscaler? we'd like to rent the hardware to check it out

**Comment by rajeshgangireddy** [comment]
> @rajeshgangireddy oh wow this is EXCITING – thank you so much! If it's not too much trouble to ask, do you know if this CPU or equivalent is available though some hyperscaler? we'd like to rent the hardware to check it out

Hi @jandom 
I will take a look into this. I will discuss with a few teammates around to arrange a machine with B70 dGPU for you.
